core/src/main/java/org/springframework/security/oauth2/server/authorization/web/OAuth2TokenEndpointFilter.java

```diff
@@ -86,8 +84,10 @@
 	private final AuthenticationManager authenticationManager;
 	private final OAuth2AuthorizationService authorizationService;
 	private final RequestMatcher tokenEndpointMatcher;
-	private final Converter<HttpServletRequest, Authentication> authorizationGrantAuthenticationConverter =
```

Comment:
The way I envisioned `authorizationGrantAuthenticationConverter` is to be a delegation-based implementation that simply delegates to the specific authorization grant implementation. Spring Security uses this strategy in a number of places. Take a look at `DelegatingAccessDeniedHandler` and `DelegatingOAuth2UserService` for examples.

For now, both `AuthorizationCodeAuthenticationConverter` and `ClientCredentialsAuthenticationConverter` should be both `private static class` and then create a `delegate` `Converter` that simply delegates to the appropriate impl.